### PR TITLE
Update build script to avoid bundling

### PR DESCRIPTION
**Pull request by natemoo-re**
## Changes

This significantly simplifies the `astro-scripts build` command. I noticed that the `bundle` option was behaving oddly with multiple `entryPoints` and code was being duplicated between them.

This PR's behavior is straightforward—build every `src/**/*.ts` file and don't try to bundle anything. Our output is just a transpiled version of our source files, so I've also removed the `source-map-support` package as well.

In the process, I found a few spots where `.js` extensions were missing from imports. I also updated `@astrojs/markdown-support` to use ESM because the `remark` dependencies can't be imported with `require`.

## Testing

- [x] Tests are passing
- **N/A** Tests updated where necessary

## Docs

Tooling updates only

### CODE REVIEW

**Comment by FredKSchott** [comment]
I know this is behind a bit, but +1 on getting this merged!

File: packages/create-astro/src/index.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import React from 'react';
```
Comment (matthewp) [inline packages/create-astro/src/index.tsx:1]: What's this file about?

Comment (natemoo-re) [reply packages/create-astro/src/index.tsx:1]: Oops, probably missed this in the rebase.

**Comment by natemoo-re** [comment]
@matthewp @FredKSchott any objections to releasing this with `astro@0.18.0`?

**Pull request merged**